lib/puppet/configurer.rb

```diff
@@ -204,8 +204,9 @@ def parse_fact_name_and_value_limits(object, path = [])
         path.pop
       end
     else
-      check_fact_name_length(path.join(), path.size)
-      check_fact_values_length(object)
+      name = path.join('.')
+      check_fact_name_length(name, path.size - 1) # second param is number of dots in fact name
```

Comment:
Suggest passing both the dotted name and the non-dotted name, so that we check the length of the latter, but issue a warning using the former.

```suggestion
      check_fact_name_length(name, path.join, path.size - 1) # third param is number of dots in fact name
```